utkdigitalinitiatives/utk-islandora7-drupal
# Address DIGITAL-876

**JIRA Issue**:  https://jirautk.atlassian.net/browse/DIGITAL-876

What Does this Do?
==================

This includes a new field configuration for the `page` nodes, adding a **field_parent_pid** with as a _text_ field with a type `text_plain`. This text field will relate the introduction nodes that exist throughout out digital collections. The value of this field is expected the be the PID of the collection level object, including the `:` character.

The field is then rendered to the screen as a `<span>` with an ID of **utk-parent-pid** and without any enclosed inner html. The span instead has a `data-pid` attribute that contains the target value to be used later by the jQuery script.

Instead of modifying mark.js, this fix instead is additive and includes a new jQuery script of **parentPid.js**. The script does the following:

1. Checks if `#utk-parent-pid` exists.
2. Gets PID value from **data-pid** attribute
3. Updates target _Islandora Collection Search_ option value

How Should This Be Tested?
==========================

### Prep
1. To prep your local vagrant you may need to run the following drush script. I had to do this to move the islandora_collection_search block to a region that existed within the sis theme.
`
drush block-configure --module=islandora_collection_search --delta=islandora_collection_search --region=splashsearch --weight=0`
2. `drush en utk_lib_digital_custom utk_banner utk_header`
3. `drush fl`
4. `drush fra -y` reverts features to code, adding the **field_parent_pid** field
5. `drush cc all`
6. Configure

### Test


1. Configure http://localhost:8000/collections/admin/islandora/tools/collection_search and make sure a couple of items are enabled under **Collections**
2. Content --> Add content --> Basic page
3. Title can be anything
4. For **Parent PID**, input the PID of the collection level object from _Step 1_ above 
5. View the node, the option for the collection should be selected

![image](https://user-images.githubusercontent.com/7376450/95700517-37814f80-0c15-11eb-992e-29b845c4b133.png)

Additional Notes
================

The `field_parent_pid` field configuration, display, and page node updates are included in the **utk_banner** feature.
